src/EventSubscriber/EzcontentJsonapiSubscriber.php

```diff
@@ -61,10 +76,33 @@ public function onResponse(FilterResponseEvent $event) {
     }
   }
 
+  /**
+   * Invalidate block cache for content listing block.
+   *
+   * We need to invalidate the block cache for content listing block
+   * because the response get cached for each jsonapi request and
+   * since we are customising to add pager using jsonapi resource with
+   * extra arguments as page that is not part of original route argument,
+   * hence it cache data for that resource irrespective of page, in-order
+   * to get result based on page number we need to clear cache.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\FilterResponseEvent $event
+   *   The event to process.
+   */
+  public function onRequest(Event $event) {
+    // Invalidate cache for content listing block for jsonapi resource.
+    $entity = $this->routeMatch->getParameter('entity');
+    if($entity instanceof BlockContent && $entity->bundle() === 'content_listing_component'
+      && $this->routeMatch->getRouteName() === 'jsonapi.block_content--content_listing_component.individual') {
+      Cache::invalidateTags($entity->getCacheTags());
```

Comment:
Instead of explicitly invalidating the cache tags, shall we add the cache context of `url.query_args:page` to the response object. In this way the cache will be automatically invalidated on changes to page argument.